src/freenet/support/Logger.java

```diff
@@ -216,35 +216,7 @@ public enum LogLevel {
 		public boolean matchesThreshold(LogLevel threshold) {
 			return this.ordinal() >= threshold.ordinal();
 		}
-		
-		@Deprecated
-		public static LogLevel fromOrdinal(int ordinal) {
-			for(LogLevel level : LogLevel.values()) {
-				if(level.ordinal() == ordinal)
-					return level;
-			}
-			
-			throw new RuntimeException("Invalid ordinal: " + ordinal);
-		}
 	}
-
-	@Deprecated
-	public static final int ERROR = LogLevel.ERROR.ordinal();
-
-	@Deprecated
-	public static final int WARNING = LogLevel.WARNING.ordinal();
-
-	@Deprecated
-	public static final int NORMAL = LogLevel.NORMAL.ordinal();
-
-	@Deprecated
-	public static final int MINOR = LogLevel.MINOR.ordinal();
```

Comment:
These are used from plugins. I do not think we can get rid of them earlier than the next decade, except if we manage to take up maintenance for all our plugins again.

Reply:
Yeah, we're getting to a point where fred can evolve less and less because of unmaintained plugin compatibility.

Until we have a solution for that (like introducing a shim that provides compatibility for legacy plugins), we'll have to live with that. Closing this PR.

Reply:
Yes, that’s a reality we have to live with.

The steps to move forward would be to first create an actual defined plugin API (which does not expose the full node) and when there are plugins using it and it’s proven itself to be sufficiently powerful, analyze all known plugins that do not use the new API which parts of fred they use. Then annotate all those places as `@Published(plugins=["...", "..."])` or such to track when plugin updates enable removing those limitations.

I’m pretty sure Bombe already has ideas for such an API.